### PR TITLE
feat: Improve FastAPI agentic template

### PR DESCRIPTION
**Pull request by leehuwuj**
## Summary by CodeRabbit

- **New Features**
	- Introduced default system prompts for improved prompt generation.
	- Enhanced error handling and user-friendly messaging in streaming responses.
	- Clarified capabilities of the DuckDuckGo search tool and streamlined configuration for other tools.
	- Updated default tool for the agentic feature to focus on weather tasks.

- **Bug Fixes**
	- Improved robustness of the streaming process with better error logging and graceful termination.
	- Enhanced error handling in the ChatSection component to prevent application crashes.
	- Updated file upload process for better compatibility with sandbox requirements.
	- Simplified instructions in README templates for clarity.

### CODE REVIEW

File: helpers/env-variables.ts [hunk 1]
```diff
@@ -13,6 +13,12 @@ import {
 
 import { TSYSTEMS_LLMHUB_API_URL } from "./providers/llmhub";
 
+const DEFAULT_SYSTEM_PROMPT =
+  "You are a helpful assistant who helps users with their questions.";
```
Comment (marcusschiesser) [inline helpers/env-variables.ts:17]: How about we instead remove the specific system prompts for Wikipedia and duckduckgo tool?

File: templates/types/streaming/fastapi/app/api/routers/vercel_response.py [hunk 2]
```diff
@@ -53,17 +53,27 @@ async def content_generator(
...
+                    yield output
+        except Exception:
+            logger.exception("Error in stream response")
+            error_message = (
+                "An error occurred while processing your request. Please try again."
+            )
+            yield cls.convert_text(error_message)
+        finally:
+            # Ensure event handler is marked as done even if connection breaks
+            event_handler.is_done = True
```
Comment (marcusschiesser) [inline templates/types/streaming/fastapi/app/api/routers/vercel_response.py:76]: Why was this missing before?

Comment (leehuwuj) [reply templates/types/streaming/fastapi/app/api/routers/vercel_response.py:76]: I can't remember the reason, but it looks to me like we had set the is_done flag before, but somehow forgot it when refactoring

File: helpers/env-variables.ts (hunk 1 above)
Comment (leehuwuj) [reply helpers/env-variables.ts:17]: Great idea, I think we can remove their system prompt because we don't need any specific tuning to use the tools

Comment (marcusschiesser) [reply helpers/env-variables.ts:17]: Might still be helpful to prioritize the query engine tool somehow if it's selected. E.g. with the right tool description

File: helpers/env-variables.ts [hunk 3]
```diff
@@ -466,9 +469,12 @@ const getSystemPromptEnv = (
       }
     });
 
-    const systemPrompt = toolSystemPrompt
-      ? `\"${toolSystemPrompt}\"`
-      : defaultSystemPrompt;
+    const systemPrompt =
+      "'" +
```
Comment (marcusschiesser) [inline helpers/env-variables.ts:473]: why "'" ?

Comment (leehuwuj) [reply helpers/env-variables.ts:473]: It's for multi-line env

File: .changeset/moody-mangos-punch.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"create-llama": patch
+---
+
+Improve system prompt and change agentic default tool to weather
```
Comment (marcusschiesser) [inline .changeset/moody-mangos-punch.md:5]:
```suggestion
Show streaming errors in Python, optimize system prompts for tool usage and set the weather tool as default for the Agentic RAG use case
```

**Review by marcusschiesser — APPROVED** [review]

**Pull request merged**